### ISSUE TITLE
Notice: Undefined index: CLIENT_ID

### ISSUE THREAD

**Issue by OD-n** [giansalex/lycet#595]
Al intentar enviar una guia de remision obtengo el error

**Comment by giansalex** [comment]
Debes agregor estos parametros en el archivo `.env`

https://github.com/giansalex/lycet/blob/f4f56ebc04ce6eba355c0f93b67f364e751d477f/.env#L37-L40

**Comment by OD-n** [comment]
Los parametros están incluidos, se puede deber a algo mas?
![image](https://github.com/giansalex/lycet/assets/28074529/c1cdb5bf-fc23-4975-b288-6ddca3f1f39f)

**Comment by giansalex** [comment]
Cliente no autorizado es por las credenciales, si estas usando el servicio Test de Nubefact, la factura debe tener el mismo RUC que ellos proveen

**Comment by OD-n** [comment]
No, solo puse el mismo CLIENT_ID y CLIENT_SECRET del archivo .env para probar, pero lo que quería es ejecutar sin esos datos, pero obtengo lo mismo: 
![image](https://github.com/giansalex/lycet/assets/28074529/84dc5566-6553-4f66-92b6-d4abba1c0092)

**Comment by giansalex** [comment]
verifica que el RUC de empresa que estas enviando en el XML, sea el mismo configurado en .env o empresas.json

**Comment by OD-n** [comment]
El RUC de company es el mismo
![image](https://github.com/giansalex/lycet/assets/28074529/425d0c56-96cb-4208-8ee4-377d45fc5404)

Y este es el del Json de empresas:
![image](https://github.com/giansalex/lycet/assets/28074529/7d0ab28a-9998-4557-bb3c-d29a58151af1)

**Comment by OD-n** [comment]
Acabo de revisar, al enviar una factura no tengo problemas, lo de CLIENT_ID solo aparece al enviar una guia.
[{
  "version": "2022",
  "tipoDoc": "09",
  "serie": "T001",
  "correlativo": "1",
  "observacion": "PRUEBA DE SISTEMAS",
  "fechaEmision": "2023-01-13T11:21:12-05:00",
  "company": {
      "ruc": "20495810250",
      "razonSocial": "GREENTER SAC",
      "nombreComercial": "GREENTER"
  },
  "destinatario": {
      "tipoDoc": "6",
      "numDoc": "20546687668",
      "rznSocial": "COMPANY SAC"
  },
  "envio": {
      "codTraslado": "01",
      "desTraslado": "VENTA",
      "modTraslado": "01",
      "pesoTotal": 10,
      "undPesoTotal": "KGM",
      "fecTraslado": "2023-04-31T13:21:12-05:00",
      "transportista": {
          "tipoDoc": "6",
          "numDoc": "20000000002",
          "rznSocial": "TRANSPORTES S.A.C",
          "nroMtc": "0001"
      },
      "llegada": {
          "ubigueo": "150203",
          "direccion": "AV. ITALIA 459"
      },
      "partida": {
          "ubigueo": "150203",
          "direccion": "AV. ITALIA 458"
      }
  },
  "details": [
      {
          "codigo": "PROD1",
          "descripcion": "PRODUCTO 1",
          "unidad": "ZZ",
          "cantidad": 2
      }
  ]
}](url)

**Comment by OD-n** [comment]
Lo descargare nuevamente, no estoy seguro de por que solo aparece ese error al intentar enviar la guia de remision porque al enviar una factura no hay problema.

**Comment by giansalex** [comment]
Hola, revise de nuevo, para pruebas de guia de remision con el servicio de Nubefact, tienes que usar el RUC: `20161515648`, de lo contrario te aparecera el error: _Cliente no autorizado_. En el archivo `empresas.json` tambien debe usarse el mismo ruc `20161515648`

**Comment by giansalex** [comment]
> Lo descargare nuevamente, no estoy seguro de por que solo aparece ese error al intentar enviar la guia de remision porque al enviar una factura no hay problema.

Es porque las guias de remison ahora se envían a un servicio diferente que las facturas.

**Comment by augustosamame** [comment]
Ok, me queda más claro. Entonces el servicio beta de las guías de remisión se hace a través de Nubefact, y no de SUNAT por intermedio de las env vars:

AUTH_URL
API_URL
CLIENT_ID
CLIENT_SECRET

Y para la versión beta, solo funciona el RUC: 20161515648

Cuáles serían los valores correctos para producción de guías de remisión entonces?
En este enlace no aparecen estos valores: https://greenter.dev/production/

**Comment by giansalex** [comment]
@augustosamame 

```
AUTH_URL=https://api-seguridad.sunat.gob.pe/v1
API_URL=https://api-cpe.sunat.gob.pe/v1
```

Los parametros `CLIENT_ID` y `CLIENT_SECRET` se obtiene en el portal de SUNAT con tu RUC.
Se usa el servicio beta de Nubefact porque SUNAT no ha habilitado uno.

**Comment by OD-n** [comment]
Acabo de enviar el json siguiente 
![image](https://github.com/giansalex/lycet/assets/28074529/5ccdf474-550b-4097-b2a5-2936e1ccc23c)

Agregue a la lista en empresas.json 
![image](https://github.com/giansalex/lycet/assets/28074529/b31959ac-bbff-4ffb-b80a-71a1118a59b8)

Y tambien un directorio con el el certificado de prueba y logo
![image](https://github.com/giansalex/lycet/assets/28074529/2262437c-7c7e-4181-9e1d-6df125d71546)

 Y aun asi tengo el mismo mensaje de error de CLIENT_ID
 
![image](https://github.com/giansalex/lycet/assets/28074529/562757df-62ef-4e30-9614-7a85226c03da)

**Comment by giansalex** [comment]
@OD-n en `empresas.json` tambien debe incluir los camps `CLIENT_ID` y `CLIENT_SECRET`
